### PR TITLE
Use Conan for External Dependencies

### PR DESCRIPTION
**Pull request by johannes-wolf**
See #40.

HUGE CI build performance increase on Windows!
Using conan for dependencies is still optional! The CI uses conan for faster builds.

- [x] Pull RocksDB via Conan
- [x] Get Conan running on the CI
- [x] Pull dependencies via Conan
- [x] Speed up the CI build
- [x] Provide a conan package

This PR improves the build times by using Ninja instead of make (with -j1) + sccache
(with filled caches, build takes < 10min).

**EDIT:**
RocksDB (6.*) Conan 2 support got merged… let's wait for a more recent version and add it in a follow-up PR, see #46.

**EDIT 2:**
RocksDB (8.8.1) got merged to conan.io.

**EDIT 3:**
Uses RocksDB 9.1.0.

**NOTE**
We can not use Catch2 from conan.io because it is broken: https://github.com/conan-io/conan-center-index/issues/13993

### CODE REVIEW

**Comment by johannes-wolf** [comment]
@josephbirkner Do you know which version of OpenSSL is needed (and why it worked without pulling any before)?

**Comment by josephbirkner** [comment]
@johannes-wolf OpenSSL 1.1.1 or 3.0 (probably better) is needed. They are currently installed as part of the cmake github action workflow...

**Comment by johannes-wolf** [comment]
@josephbirkner Is linking OpenSSL static ok? I could not get the `@rpath` on macOS right… :(

**Comment by johannes-wolf** [comment]
The tests keep timing out on Windows (this issue existed before).

**Comment by josephbirkner** [comment]
> @josephbirkner Is linking OpenSSL static ok? I could not get the `@rpath` on macOS right… :(

Yes 👍

**Review by josephbirkner — APPROVED** [review]
Awesome 🥳 But: Once we merge, we will now need to introduce conan2 into erdblick and prepare for interactions between conan2 and the emscripten cmake toolchain. So I'd say, let's talk about how we do this on monday @MisterGC .

**Comment by MisterGC** [comment]
> Awesome 🥳 But: Once we merge, we will now need to introduce conan2 into erdblick and prepare for interactions between conan2 and the emscripten cmake toolchain. So I'd say, let's talk about how we do this on monday @MisterGC .

Because of this we will delay merging this one to the period after the upcoming release (0.4.0).

**Comment by johannes-wolf** [comment]
With the change to `conanfile.py` Windows builds fail because Conan 2 seems to generate unescaped paths on Windows :sob:.

**Comment by johannes-wolf** [comment]
@josephbirkner I moved the dependencies to `deps.cmake` and re-added `FetchContent` (+ the find_package integration) to not break consumers.

File: deps.cmake [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+set(WITH_BENCHMARK_TOOLS NO CACHE BOOL "rocksdb without benchmarking")
+set(BENCHMARK_ENABLE_GTEST_TESTS NO CACHE BOOL "rocksdb without gtest")
+set(DISABLE_WARNING_AS_ERROR 1 CACHE BOOL "rocksdb warnings are ok")
+FetchContent_Declare(rocksdb
+  GIT_REPOSITORY "https://github.com/facebook/rocksdb.git"
+  GIT_TAG        dc87847e65449ef1cb6f787c5d753cbe8562bff1 # Use version greater than v8.6.7 once released.
+  GIT_SHALLOW    OFF
+  FIND_PACKAGE_ARGS NAMES RocksDB)
+
+FetchContent_Declare(stx
```
[the remark this answers is not in the document]
Comment (johannes-wolf) [reply deps.cmake]: After that PR got merged.

**Comment by johannes-wolf** [comment]
Waiting for simfil/0.1.2 conan recipe merge.

**Comment by johannes-wolf** [comment]
Rebased + updated for simfil 0.1.3.

**Comment by johannes-wolf** [comment]
Rebased on `main` and updated the RocksDB dependency to v9+, which fixes the non-conan build.

**Review by josephbirkner — APPROVED** [review]
Looks great, just waiting for simfil 2.1.0 to be merged into conan-center-index. See https://github.com/conan-io/conan-center-index/pull/24393

**Comment by johannes-wolf** [comment]
Rebased (+ simfil 0.2.1)

File: deps.cmake [hunk 2]
```diff
@@ -65,17 +65,25 @@ else()
...
-      GIT_TAG        "v9.1.0"
-      GIT_SHALLOW    OFF)
-    FetchContent_MakeAvailable(RocksDB)
-    add_library(RocksDB::rocksdb ALIAS rocksdb)
+    block()
+      set(WITH_GFLAGS NO CACHE BOOL "rocksdb without gflags")
+      set(WITH_TESTS NO CACHE BOOL "rocksdb without tests")
+      set(WITH_BENCHMARK_TOOLS NO CACHE BOOL "rocksdb without benchmarking")
+      set(BENCHMARK_ENABLE_GTEST_TESTS NO CACHE BOOL "rocksdb without gtest")
+      set(WITH_TOOLS NO CACHE BOOL "rocksdb without tools")
```
Comment (johannes-wolf) [inline deps.cmake:73]: I guess we do not need to build the tools?

**Pull request merged**